genesyscloud/webdeployments_configuration/resource_genesyscloud_webdeployments_configuration_test.go

```diff
@@ -1043,8 +1067,9 @@ func generateWebDeploymentConfigCobrowseSettings(cbEnabled, cbAllowAgentControl
 		channels = [ %s ]
 		mask_selectors = [ %s ]
 		readonly_selectors = [ %s ]
+		pause_criteria = [ %s ]
 	}
```

Comment:
We define items in a schema list like this: 

```golang
pause_criteria { // item one in the list
  url_fragment = ""
  condition   = ""
}

pause_criteria { // item two 
  url_fragment = ""
  condition   = ""
}
```

Here is an example of we usually use functions to Generate list items - https://github.com/MyPureCloud/terraform-provider-genesyscloud/blob/main/genesyscloud/outbound_contact_list_contact/resource_genesyscloud_outbound_contact_list_contact_utils.go#L144

We have a parent `generateXYZ` resource whose final parameter is `nestedBlocks ...string,`. This allows you to pass as many calls to a function like `generatePauseCriteria`, that will generate the nested schema objects.

e.g.

```golang
var config = generateWebdeploymentConfigResource(
  resourceId,
  name,
  generatePauseCriteria(
    "url 1",
    "condition 1",
  ),
  generatePauseCriteria(
    "url 2",
    "condition 2",
  ),
)
```

How you have done it might be syntactically correct but it's a pretty big departure from the convention in this project, so it would be good to align it.

Reply:
you want me to do something like 

func generateWebDeploymentConfigCobrowseSettings(cbEnabled, cbAllowAgentControl string, cbAllowAgentNavigation string, cbChannels []string, cbMaskSelectors []string, cbReadonlySelectors []string, pauseCriteriaBlocks ...string,) string {

	return fmt.Sprintf(`
	cobrowse {
		enabled = %s
		allow_agent_control = %s
		allow_agent_navigation = %s
		channels = [ %s ]
		mask_selectors = [ %s ]
		readonly_selectors = [ %s ]
		%s
	}
`, cbEnabled, cbAllowAgentControl, cbAllowAgentNavigation, strings.Join(cbChannels, ", "), strings.Join(cbMaskSelectors, ", "), strings.Join(cbReadonlySelectors, ", "), strings.Join(pauseCriteriaBlocks, "\n"))
}

func generatePauseCriteria(urlFragment, condition string) string {
	return fmt.Sprintf(`pause_criteria {
	url_fragment = "%s"
	condition = "%s"
}`, urlFragment, condition)
}
?

Reply:
Exactly 👍